discovery-provider/src/tasks/index_nethermind.py

```diff
@@ -220,29 +227,34 @@ def is_block_on_chain(web3: Web3, block: Block):
     Determines if the provided block is valid on chain by fetching its hash.
     """
     try:
-        block_from_chain = web3.eth.get_block(block.blockhash)
+        block_from_chain = web3.eth.get_block(cast(HexStr, block.blockhash))
         return block_from_chain
     except BlockNotFound:
         return False
     # Raise any other type of exception
 
 
 def get_next_block(web3: Web3, latest_database_block: Block, final_poa_block=0):
+    if latest_database_block.number is None:
+        logger.info(f"Block number invalid {latest_database_block}, returning early")
+        return False
+
     # Get next block to index
     next_block_number = latest_database_block.number - (final_poa_block or 0) + 1
     try:
         next_block = web3.eth.get_block(next_block_number)
-        next_block = AttributeDict(
-            next_block, number=next_block.number + final_poa_block
-        )
-        return next_block
+        next_block_mut = dict(next_block)
```

Comment:
isn't `get_block` already returning `BlockData` which is a typed dict? Do we need this `dict()` call and subsequent cast?

Reply:
yep, can make the change! i think i had added this in some intermediate state and didn't undo.

Reply:
Just kdding -- this was necessary. BlockData is just a TypedDict. It says nothing about immutability, but the return type of get_block is an AttributeDict from the web3 library, which is in fact immutable, causing a runtime error. Python is awful!

Reply:
😢 sad!